From ee36a284641d9ac2b412bba3738729fd5b969f42 Mon Sep 17 00:00:00 2001
From: Alyssa Ross <hi@alyssa.is>
Date: Sun, 1 Dec 2019 22:11:39 +0000
Subject: [PATCH 5/9] libbrillo: fix build with relative platform2_root

---
 libbrillo/BUILD.gn | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libbrillo/BUILD.gn b/libbrillo/BUILD.gn
index 6739def9c..75b6da6f2 100644
--- a/libbrillo/BUILD.gn
+++ b/libbrillo/BUILD.gn
@@ -360,7 +360,8 @@ shared_library("libpolicy-${libbase_ver}") {
     "openssl",
     "protobuf-lite",
   ]
-  ldflags = [ "-Wl,--version-script,${platform2_root}/libbrillo/libpolicy.ver" ]
+  libpolicy_ver = rebase_path("//libbrillo/libpolicy.ver")
+  ldflags = [ "-Wl,--version-script,${libpolicy_ver}" ]
   sources = [
     "policy/device_policy.cc",
     "policy/device_policy_impl.cc",
-- 
2.23.0

